### PR TITLE
OdcmModel Refinements

### PR DESCRIPTION
**Pull request by tonycrider**
[1] Remove OdcmField, consolidating relevant properties into OdcmProperty.
[2] Fix OdcmClass Key property to recursively find Key properties in base classes.
[3] Removed the unused "AsReference" methods.
[4] Miscellaneous clean-up and more ODataReader unit tests.

### CODE REVIEW

**Comment by Sridhar-MS** [comment]
LGTM.

File: src/Core/Vipr/ODataWriter.cs [hunk 1]
```diff
@@ -9,21 +7,19 @@
...
+    using Vipr.Core.CodeModel;
+    using System;
+    
     /// <summary>
     /// Class to produce the template output
     /// </summary>
-
-#line 1 "E:\gittf\Roxel\O365\Public\src\EdmDecomposition\EdmDecomposition\ODataWriter.tt"
+    
+    #line 1 "C:\Git\tonycrider\vipr\src\Core\Vipr\ODataWriter.tt"
```
Comment (piotrpMSFT) [inline src/Core/Vipr/ODataWriter.cs]: This looks like a reference to a developer's local file system...

File: src/Core/Vipr/ODataWriter.cs [hunk 2]
```diff
@@ -1,6 +1,4 @@
-﻿// Copyright (c) Microsoft. All rights reserved.
-// Licensed under the MIT license. See LICENSE file in the project root for full license information.
-// ------------------------------------------------------------------------------
+﻿// ------------------------------------------------------------------------------
 // <auto-generated>
```
Comment (piotrpMSFT) [inline src/Core/Vipr/ODataWriter.cs]: Do we still need ODataWriter.cs?

File: src/Writers/CSharpWriter/CSharpWriter.Code.cs [hunk 3]
```diff
@@ -1029,7 +1029,7 @@ private void Write(EntityContainerConstructor entityContainerConstructor)
             _("partial void OnContextCreated();");
         }
 
-        public void Write(IEnumerable<OdcmField> fields)
```
Comment (piotrpMSFT) [inline src/Writers/CSharpWriter/CSharpWriter.Code.cs]: This should just be removed. The CSharp model included Odcm fields for completeness but did not take dependencies on the fields existing, IIRC.

File: src/Writers/CSharpWriter/CSharpWriter.Code.cs [hunk 4]
```diff
@@ -1037,7 +1037,7 @@ public void Write(IEnumerable<OdcmField> fields)
             }
         }
 
-        public void Write(OdcmField odcmField)
+        public void Write(OdcmProperty odcmField)
```
Comment (piotrpMSFT) [inline src/Writers/CSharpWriter/CSharpWriter.Code.cs]: This should just be removed. The CSharp model included Odcm fields for completeness but did not take dependencies on the fields existing, IIRC.

**Comment by piotrpMSFT** [comment]
Some comments, LGTM otherwise.

**Pull request merged**